tools/trap/trap.c

```diff
@@ -350,32 +421,35 @@ int main(int argc, char *argv[])
 	printf("Target device locked and ready to DUMP!!!\n");
 
 	while (1) {
-		printf("Choose from the following options:-\n1. RAM Dump\n2. Userfs Dump\n3. Both RAM and Userfs dumps\n4. External Userfs Partition Dump\n5. Exit TRAP Tool\n6. Reboot TARGET Device\n");
+		printf("Choose from the following options:-\n1. RAM Dump\n2. Userfs Dump\n3. Both RAM and Userfs dumps\n4. External Userfs Partition Dump\n5. Dump specific file from target device\nr. Reboot TARGET Device\nx. Exit TRAP tool\n");
 		fflush(stdout);
-		scanf("%d", &choice);
+		scanf("%c", &choice);
 		getchar();
 
 		DUMP_FLAGS = CLEAR_DUMP_FLAGS;
 
 		switch (choice) {
-		case 1:
+		case '1':
 			DUMP_FLAGS |= RAMDUMP_FLAG;
 			break;
-		case 2:
+		case '2':
 			DUMP_FLAGS |= USERFSDUMP_FLAG;
 			break;
-		case 3:
+		case '3':
 			DUMP_FLAGS |= RAMDUMP_FLAG;
 			DUMP_FLAGS |= USERFSDUMP_FLAG;
 			break;
-		case 4:
+		case '4':
 			DUMP_FLAGS |= EXTUSERFS_DUMP_FLAG;
 			break;
-		case 5:
+		case '5':
+			DUMP_FLAGS |= FILE_DUMP_FLAG;
```

Comment:
Does this support multiple command at once? If not, we can just set it without bit comparision.

Reply:
Does not support currently,
We can do it
I will add this in a separate commit.